nut 2.8.0 build failure on MacOS

👋 trying to build the latest release, but run into some build issue. The error log is as below:


<details>
<summary>build error</summary>

```
dummy-ups.c:238:24: error: no member named 'st_mtim' in 'struct stat'
                                        if (datafile_stat.st_mtim.tv_sec != fs.st_mtim.tv_sec) {
                                            ~~~~~~~~~~~~~ ^
dummy-ups.c:238:45: error: no member named 'st_mtim' in 'struct stat'
                                        if (datafile_stat.st_mtim.tv_sec != fs.st_mtim.tv_sec) {
                                                                            ~~ ^
2 errors generated.
```

</details>

full build log, https://github.com/Homebrew/homebrew-core/actions/runs/2287073413
relates to Homebrew/homebrew-core#101035

That would be MacOS? Which version? Hope I can find some clues in public
headers. Maintaining compat with obscure proprietary systems is hard
guesswork :\

Aldernately, can you locate and experiment how tge file modification time
field is called in `struct stat`, if present?

On Sat, May 7, 2022, 20:31 Rui Chen ***@***.***> wrote:

> 👋 trying to build the latest release, but run into some build issue. The
> error log is as below:
> build error
>
> dummy-ups.c:238:24: error: no member named 'st_mtim' in 'struct stat'
>
>                                         if (datafile_stat.st_mtim.tv_sec != fs.st_mtim.tv_sec) {
>
>                                             ~~~~~~~~~~~~~ ^
>
> dummy-ups.c:238:45: error: no member named 'st_mtim' in 'struct stat'
>
>                                         if (datafile_stat.st_mtim.tv_sec != fs.st_mtim.tv_sec) {
>
>                                                                             ~~ ^
>
> 2 errors generated.
>
>
> full build log,
> https://github.com/Homebrew/homebrew-core/actions/runs/2287073413
> relates to Homebrew/homebrew-core#101035
> <https://github.com/Homebrew/homebrew-core/pull/101035>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/networkupstools/nut/issues/1415>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMPTFAVT6ZFPLN5EHXNZUDVI2ZHRANCNFSM5VKYD6NA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

--- a/drivers/dummy-ups.c
+++ b/drivers/dummy-ups.c
@@ -235,7 +235,7 @@ void upsdrv_updateinfo(void)
 					memset(&datafile_stat, 0, sizeof(struct stat));
 					next_update = 1;
 				} else {
-					if (datafile_stat.st_mtim.tv_sec != fs.st_mtim.tv_sec) {
+					if (datafile_stat.st_mtime != fs.st_mtime) {
 						upsdebugx(2, "upsdrv_updateinfo: input file was already read once to the end, but changed later - re-reading");
 						/* updated file => retry ASAP */
 						next_update = 1;

